src/DynamoPackages/PackageLoader.cs

```diff
@@ -297,6 +294,7 @@ private void TryLoadPackageIntoLibrary(Package package)
 
                 package.LoadState.SetAsLoaded();
                 PackgeLoaded?.Invoke(package);
+                PackagesLoaded?.Invoke(loadedAssemblies);
```

Comment:
Can you explain how moving this call into `TryLoadPackageIntoLibrary` fixes the bug?

Reply:
agree - I'm not clear on how this works.

Reply:
There are conditions in the `TryLoadPackageIntoLibrary` that (if evaluated as true) will skip loading the package.
Ex: If the package was already loaded or if it was marked as unloaded, or if an exception is thrown...

In the case where we do not load a package then we should not invoke the `PackagesLoaded` event for that package.

In the unit test I wrote (for example) the `PackagesLoaded` event is not called if the package was already loaded